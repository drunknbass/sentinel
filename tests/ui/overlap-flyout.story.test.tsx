import { describe, it, expect } from 'vitest'
import { render } from '@testing-library/react'
import OverlapFlyout from '@/components/overlap-flyout'

const sampleItems = Array.from({ length: 3 }).map((_, i) => ({
  incident_id: `RG2400${i}`,
  call_type: ['Armed Robbery','Traffic Collision','Suspicious Activity'][i % 3],
  call_category: 'other',
  priority: [10,50,80][i % 3],
  received_at: new Date().toISOString(),
  address_raw: '123 MAIN ST',
  area: 'RIVERSIDE',
  disposition: 'Active',
  lat: 33.9533,
  lon: -117.3962,
}))

describe('OverlapFlyout (storybook)', () => {
  it('renders primary state', () => {
    const { container } = render(
      <div style={{ position:'relative', width: 320, height: 240 }}>
        <OverlapFlyout
          items={sampleItems}
          lat={33.9533}
          lon={-117.3962}
          anchor={{ x: 100, y: 90 }}
          onSelect={() => {}}
          onClose={() => {}}
        />
      </div>
    )
    expect(container).toMatchSnapshot()
  })
})

