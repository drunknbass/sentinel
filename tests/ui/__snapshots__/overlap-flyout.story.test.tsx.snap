// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OverlapFlyout (storybook) > renders primary state 1`] = `
<div>
  <div
    style="position: relative; width: 320px; height: 240px;"
  >
    <div
      class="cyber-flyout"
      style="position: absolute; left: 120px; top: 90px; transform: translate(0,-50%); z-index: 5000;"
    >
      <div
        class="cyber-flyout-inner"
      >
        <div
          class="cyber-flyout-header"
        >
          <span>
            3
             INCIDENTS @ 
            33.9533
            , 
            -117.3962
          </span>
          <button
            class="cyber-close"
          >
            Ã—
          </button>
        </div>
        <div
          class="cyber-flyout-list"
        >
          <button
            class="cyber-item"
          >
            <span
              class="badge"
              style="background: rgb(239, 68, 68);"
            />
            <span
              class="id"
            >
              RG24000
            </span>
            <span
              class="type"
            >
              Armed Robbery
            </span>
            <span
              class="area"
            >
              RIVERSIDE
            </span>
          </button>
          <button
            class="cyber-item"
          >
            <span
              class="badge"
              style="background: rgb(234, 179, 8);"
            />
            <span
              class="id"
            >
              RG24001
            </span>
            <span
              class="type"
            >
              Traffic Collision
            </span>
            <span
              class="area"
            >
              RIVERSIDE
            </span>
          </button>
          <button
            class="cyber-item"
          >
            <span
              class="badge"
              style="background: rgb(132, 204, 22);"
            />
            <span
              class="id"
            >
              RG24002
            </span>
            <span
              class="type"
            >
              Suspicious Activity
            </span>
            <span
              class="area"
            >
              RIVERSIDE
            </span>
          </button>
        </div>
      </div>
      <style>
        
        .cyber-flyout{pointer-events:auto}
        .cyber-flyout-inner{width:248px;background:rgba(0,0,0,0.92);border:2px solid #ffb000;box-shadow:0 0 16px rgba(255,176,0,0.6), inset 0 0 12px rgba(255,176,0,0.2);font-family:"IBM Plex Mono","Courier New",monospace;color:#ffb000}
        .cyber-flyout-header{display:flex;align-items:center;justify-content:space-between;padding:8px 10px;font-weight:bold;letter-spacing:.5px;background:linear-gradient(90deg, rgba(255,176,0,0.15), rgba(0,0,0,0))}
        .cyber-close{color:#ffb000;background:transparent;border:none;font-size:18px;cursor:pointer}
        .cyber-flyout-list{display:flex;flex-direction:column;max-height:180px;overflow:auto}
        .cyber-item{display:grid;grid-template-columns:10px 1fr;grid-template-rows:auto auto;gap:4px 8px;text-align:left;padding:8px 10px;background:transparent;border:none;color:#ffb000;cursor:pointer}
        .cyber-item:hover{background:rgba(255,176,0,0.08)}
        .cyber-item .badge{grid-row:span 2;width:10px;height:100%;display:block}
        .cyber-item .id{font-size:11px;opacity:.75}
        .cyber-item .type{font-size:12px;font-weight:700;letter-spacing:.3px}
        .cyber-item .area{font-size:11px;opacity:.7}
        .cyber-note{padding:6px 10px;font-size:11px;opacity:.7}
      
      </style>
    </div>
  </div>
</div>
`;
